runtime/compiler/wasm/reader.go

```diff
@@ -158,6 +158,16 @@ func (r *WASMReader) readSection() error {
 			return err
 		}
 
+	case sectionIDStart:
+		if r.Module.StartFunctionIndex != nil {
```

Comment:
Why not moving this check to `readStartSection`?

So that this case will become easier to test.

Reply:
I structured the code in a way that each section reader function only handle reading a section, independent of the whole module structure, and then have this function ensure there are no duplicates.